ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1306,6 +1308,122 @@ public void shouldFailTerminateOnInvalidDeleteTopicPattern() {
     ksqlResource.terminateCluster(request);
   }
 
+  @Test
+  public void shouldFailIfRegisterTopicAlreadyExists() {
+    // Given:
+    final String registerSql = "REGISTER TOPIC FOO WITH (kafka_topic='bar', value_format='json');";
+    metaStore.putTopic(new KsqlTopic("FOO", "foo", new KsqlJsonTopicSerDe(), true));
+
+    when(commandStore.enqueueCommand(any(), any(), any(), any()))
+        .thenThrow(new IllegalArgumentException("Should never get to the enqueue stage!"));
+
+    // the below (commented) statement should cause this test to pass without the direct
+    // topic put on the meta store, but it does not. FIXME(agavra)
+    // makeSingleRequest(registerSql, CommandStatusEntity.class);
+
+    // Then:
+    expectedException.expect(KsqlRestException.class);
+    expectedException.expect(exceptionStatusCode(is(Code.BAD_REQUEST)));
+    expectedException.expect(exceptionErrorMessage(
+        errorMessage(is("Topic already registered: FOO"))));
+
+    // When:
+    makeSingleRequest(registerSql, CommandStatusEntity.class);
+  }
+
+  @Test
+  public void shouldFailIfCreateExistingSourceStream() {
+    // Given:
+    final Schema schema = SchemaBuilder.struct().field("val", Schema.OPTIONAL_STRING_SCHEMA);
```

Comment:
As above, rather than have a test building schemas, (which really isn't pertinent to the test), why not just use mocks? Or at least make `schema` a constant/field and move its creation out of the test cases.

Reply:
I'll put it in a constant field but I think it's overkill to mock Schema since that's not a heavy component and is not defined within KSQL.